docs/proposals/proposal-002-run.md

```diff
@@ -164,108 +165,134 @@ change are understandable. This may include manifests or workflow examples
+#### Use Case 2: Benchmark manifest is defined in a _different_ repository
 
-The benchmark workflow which contains the benchmark jobs and their test steps may look like the following:
+We want to accommodate different methods of setting up the tests depending on the CNCF project. Given this, the benchmark manifest could be defined in a different repository. In this case, the `k8s_manifest_url` would be, for example, `https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/e93136094735c1a52cbbef3d7e362839f26f4944/benchmark-tests/falco-benchmark-tests.yaml`.
 
-```yaml
-# .github/workflows/tests/falco-benchmark-workflow.yaml
-jobs:
-  stress-ng-test:
-    runs-on: ubuntu-latest
-    steps:
-      - run: |
-          # the action to take here depends on the Functional Unit of the CNCF project. wait for amount of time, for resources
-          kubectl apply -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
-          # the one above is a workflow with a single benchmark job, but if your workflow needs multiple benchmark job, it is enough to define additional steps:
-          # e.g. for redis-test: kubectl apply -f https://github.com/falcosecurity/cncf-green-review-testing/blob/main/kustomize/falco-driver/ebpf/redis.yaml
-          # for event-generator-test -> kubectl apply -f https://github.com/falcosecurity/cncf-green-review-testing/blob/main/kustomize/falco-driver/ebpf/falco-event-generator.yaml
-          wait 15m
-      - delete: |
-         kubectl delete -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml  # other Falco tests:
-```
+Applying manifests from a different repository not controlled by Green Reviews is a potential security risk. See next section.
 
-The benchmark job has some test instructions/steps. In this case, it applies an upstream Kubernetes manifest. This manifest contains a `while` loop that runs `stress-ng`. The manifest already defines where the test should run in the cluster i.e. in which namespace. The functional unit test is time-bound in this case and scoped to 15 minutes. Therefore, we deploy this test, wait for 15 minutes, then delete the manifest to end the loop. The test steps depend on the functional unit of each CNCF project.
+### Versioning / Security
 
-In the example above, the Kubernetes manifest that is applied to the cluster is located in a different repository: this is the case of an externally defined benchmark
-
-Each workflow should ensure that any artefacts that were deployed as part of the benchmark job should be deleted at the end of the test run.
+Manifests in `project.json` are pinned to a Git commit SHA rather than a branch such as `main`. This mitigates the risk that a malicious workload could be included in the benchmark manifest and ensures that any changes to the manifests are reviewed by one of the Green Reviews maintainers.
```

Comment:
@leonardpahlke @nikimanoledaki @dipankardas011 FYI I added this note on pinning manifests to SHAs following our discussion in last weeks meeting.

LMK if more detail is needed

Reply:
I think we should add some seccomp profiles, quota limits on the namespace we are dealing with so that we don't have impact on other monitoring workloads, also we should perform some sort of image scaning before begining to start benchmarking to safeguard from any potential problems

anyways let me know your thoughts :)
I was just checking out CIS benchmark for any k8s cluster

Reply:
@dipankardas011 Those are good recommendations for securing the cluster but I think we should keep this focused on the pipeline.

The future of the cluster is uncertain. If we only provision on-demand the design is likely to be different to a permanent cluster even a single node one. So I would not invest time until the design is clear.

However your approach of being guided by the CIS benchmark is sound. It's also a good point that by running both the project components and the benchmarks in the `benchmark` namespace we can restrict its access.